File: src/main/java/dev/budget/reconciler/guice/ElasticSearchModule.java
package dev.budget.reconciler.guice;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
import org.elasticsearch.client.Client;
import org.elasticsearch.node.Node;
import org.elasticsearch.node.NodeBuilder;

public class ElasticSearchModule extends AbstractModule {

	protected void configure() {}

	@Provides
	@Singleton
	private Node createNode() {
		// Create a local node that persists data to 'data/' between restarts
		return NodeBuilder.nodeBuilder()
				.clusterName("mint-ynab-reconciler")
				.local(true)
				.data(true)
				.loadConfigSettings(false)
				.build();
	}

	@Provides
	@Singleton
	private Client createClient(Node node) {
		return node.client();
	}
}

File: src/main/scala/dev/budget/reconciler/modules/ElasticSearchModule.scala
package dev.budget.reconciler.modules

import dev.budget.reconciler.dao.ESTransactionDao
import dev.budget.reconciler.es.{ElasticSearchAdmin, ManagedElasticSearch}
import dev.budget.reconciler.health.ElasticSearchHealth
import org.elasticsearch.client.Client
import org.elasticsearch.node.{Node, NodeBuilder}
import scaldi.Module

class ElasticSearchModule extends Module {

  bind [Node] to {
    NodeBuilder.nodeBuilder()
      .clusterName("mint-ynab-reconciler")
      .local(true)
      .data(true)
      .loadConfigSettings(false)
      .build()
  }

  bind [Client] to inject[Node].client()

  bind [ManagedElasticSearch] to new ManagedElasticSearch
  bind [ElasticSearchAdmin] to new ElasticSearchAdmin
  bind [ElasticSearchHealth] to new ElasticSearchHealth

  bind [ESTransactionDao] to new ESTransactionDao
}
